src/net.h

```diff
@@ -505,6 +506,10 @@ class CNode
      * peer eviction criterium in CConnman::AttemptToEvictConnection. */
     std::atomic<std::chrono::seconds> m_last_block_time{0s};
 
+    /** UNIX epoch time of the last block announcement from this peer that we had
+     * not yet seen from any peer. Used for potential stale-tip eviction. */
+    std::atomic<std::chrono::seconds> m_last_block_announcement{0s};
```

Comment:
I don't think this belongs in `CNode`. It is used in our outbound eviction logic which is part of net processing and it isn't used in net at all (imo, the eviction logic should be encapsulated in its own module entirely #25268 😉).

Also afaict, there is no need to move `m_last_block_announcement`, because you can just return it in `PeerManager::GetNodeStateStats`.